File: abc113b.cs
using System;

public class ABC113B {
  static public void Main(){
    string[] buf;
    buf = Console.ReadLine().Split(' ');
    int N = int.Parse(buf[0]);
    buf = Console.ReadLine().Split(' ');
    int T = int.Parse(buf[0]);
    int A = int.Parse(buf[1]);
    buf = Console.ReadLine().Split(' ');
    int[] H = new int[buf.Length];
    for (int i = 0; i < buf.Length; i++) {
      H[i] = int.Parse(buf[i]);
    }
    double closestDiff = double.PositiveInfinity;
    int closestIdx = -1;
    for (int i = 0; i < H.Length; i++) {
      double diff = Math.Abs(A - (T - H[i] * 0.006));
      if (diff < closestDiff) {
        closestIdx = i;
        closestDiff = diff;
      }
    }
    Console.WriteLine(closestIdx + 1);
  }
}

File: abc113b.py

def main():
    buf = input()
    buflist = buf.split()
    N = int(buflist[0])
    buf = input()
    buflist = buf.split()
    T = int(buflist[0])
    A = int(buflist[1])
    buf = input()
    buflist = buf.split()
    H = list(map(int, buflist))

    mindiff = float("inf")
    point = None
    for i in range(N):
        temp = T - (H[i] * 0.006)
        diff = abs(A - temp)
        if diff < mindiff:
            mindiff = diff
            point = i
    print(point+1)

if __name__ == '__main__':
    main()
